flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializerTestSchemas.java

```diff
@@ -0,0 +1,252 @@
+                                .setType("FLOAT")
+                                .setMode("REQUIRED"),
+                        new TableFieldSchema()
+                                .setName("species")
+                                .setType("STRING")
+                                .setMode("REQUIRED"),
+                        new TableFieldSchema()
+                                .setName("flighted")
+                                .setType("BOOLEAN")
+                                .setMode("REQUIRED"),
+                        new TableFieldSchema()
+                                .setName("sound")
+                                .setType("BYTES")
+                                .setMode("REQUIRED"),
+                        new TableFieldSchema()
+                                .setName("required_record_field")
+                                .setType("RECORD")
+                                .setMode("REQUIRED")
+                                .setFields(subFieldsNullable));
+        TableSchema tableSchema = new TableSchema().setFields(fields);
+        BigQuerySchemaProvider bigQuerySchemaProvider = new BigQuerySchemaProvider(tableSchema);
+        Descriptor descriptor = bigQuerySchemaProvider.getDescriptor();
+        return new BigQueryAvroToProtoSerializerTestResult(
+                bigQuerySchemaProvider.getSchema(), descriptor);
+    }
+
+    public static BigQueryAvroToProtoSerializerTestResult testLogicalTypesConversion() {
+        List<TableFieldSchema> fields =
+                Arrays.asList(
+                        new TableFieldSchema()
+                                .setName("timestamp")
+                                .setType("TIMESTAMP")
+                                .setMode("NULLABLE"),
+                        new TableFieldSchema()
+                                .setName("numeric_field")
+                                .setType("NUMERIC")
+                                .setMode("REQUIRED"),
+                        new TableFieldSchema()
+                                .setName("bignumeric_field")
+                                .setType("BIGNUMERIC")
+                                .setMode("NULLABLE"),
+                        new TableFieldSchema()
+                                .setName("geography")
+                                .setType("GEOGRAPHY")
+                                .setMode("REQUIRED"),
+                        new TableFieldSchema().setName("Json").setType("JSON").setMode("REQUIRED"));
+
+        TableSchema tableSchema = new TableSchema().setFields(fields);
+        BigQuerySchemaProvider bigQuerySchemaProvider = new BigQuerySchemaProvider(tableSchema);
+        Descriptor descriptor = bigQuerySchemaProvider.getDescriptor();
+        return new BigQueryAvroToProtoSerializerTestResult(
+                bigQuerySchemaProvider.getSchema(), descriptor);
+    }
+
+    public static BigQueryAvroToProtoSerializerTestResult testAllPrimitiveSchemaConversion() {
```

Comment:
How will this be used differently from the primitive schema provider above?

Reply:
The idea was to include any datatypes that are not obtained as a part of conversion of BQ table Schema. Removed the redundant fields.